### DISCUSSION TITLE
Webhook Integration

### DISCUSSION THREAD

**Discussion by rjdjohnston** [payloadcms/payload#524] [Feature Requests & Ideas]
Is there a roadmap item to add webhook capability? Specifically, a Zapier integration.

**Comment by DanRibbens** [comment]
Not at this time. 

I'm unaware of what is required to integrate with Zapier without doing a little research. It may already be possible, but If there are requirements that Payload is missing, I can help with that.

Ideally there would be a Payload plugin to handle this integration. If you're unfamiliar, Payload plugins are an ideal way to extend Payload apps in routine ways, this is especially great for integrations that can be shared across many projects. We can provide support to those developing plugins for the community to use, but we're not able to create plugins for each 3rd party integration without additional help.

Anyone interested in creating Payload plugins should start with the [docs](https://payloadcms.com/docs/plugins/overview) documentation. Also this is a great example of another plugin that uses hooks: https://github.com/jeanbmar/payload-s3-upload

Let me know if there are more features needed to be added to Payload to make this a reality.

**Reply by jmikrut** [reply]
To add to this, @rjdjohnston — you can currently do everything webhooks can do with Payload's [`hooks`](https://payloadcms.com/docs/hooks/overview). But Payload's hooks are _**way**_ more powerful and a better pattern than traditional webhooks which you typically have significantly less control over. 

Webhooks are a pattern that is seen for 3rd-party hosted SaaS solutions where you don't have access to the backend code, but with Payload, of course, you do and you can do whatever you need.

For this reason, as of now webhooks are not on our radar, and we are concentrating on making hooks as powerful as they can be.

Does this make sense? If you have a use case for webhooks after hearing the above, we'd be very interested in hearing about it!

**Comment by rjdjohnston** [comment]
I’ve been evaluating JAMstack CMSs for the past week or so. It’s a challenge to find a good mix of capabilities and simplicity without a huge foot print or lock-in. I’m going to evaluate payload more over the next week. 

Thanks for the thoughtful responses

**Comment by rjdjohnston** [comment]
Hi @jmikrut , Do you guys have a Discord for payload yet? I'm working through implementing. A discord community could be helpful

**Reply by DanRibbens** [reply]
I just added it to the readme. You can find the link on https://github.com/payloadcms/payload
